scripts/h5toobj.py

```diff
@@ -0,0 +1,343 @@
+        redis_host = 'localhost'
+        redis_endpoint = '{}:{}'.format(redis_host, args.redis_port)
+    else:
+        redis_host = args.redis
+        redis_endpoint = '{}:{}'.format(redis_host, args.redis_port)
+    ts = katsdptelstate.TelescopeState(redis_endpoint)
+    logger.info("Connected to Redis on %s. DB has %d existing keys",
+                redis_endpoint, len(ts.keys()))
+
+    r_str = ""
+    for attr in h5_file['TelescopeState'].attrs:
+        r_str += redis_gen_proto("SET", attr, h5_file['TelescopeState'].attrs[attr])
+    if redis_endpoint:
+        redis_bulk_str(r_str, redis_host, args.redis_port)
+
+    if not args.obj_only:
+        for d_count, dset in enumerate(h5_file['TelescopeState'].keys()):
+            st = time.time()
+            r_str = ""
+            d_val = h5_file['TelescopeState'][dset].value
+             # much quicker to read it first and then iterate
+            for (timestamp, pval) in d_val:
+                packed_ts = struct.pack('>d', float(timestamp))
+                r_str += redis_gen_proto("ZADD", str(dset), "0", packed_ts + pval)
+            bss = time.time()
+            if redis_endpoint:
+                redis_bulk_str(r_str, redis_host, args.redis_port)
+                logger.info("Added %d items in %gs to key %s. Bulk insert time: %g",
+                            len(d_val), time.time() - st, dset, time.time() - bss)
+        logger.info("Added %d ranged keys to TelescopeState", d_count + 1)
+
+    if args.redis_only and args.redis is None:
+        logger.warning("Terminating locally launched redis instance "
+                       "(also saves telstate to local dump.rdb)")
+        try:
+            cli_cmd = "/usr/bin/redis-cli -p {} SHUTDOWN SAVE".format(args.redis_port)
+            subprocess.call(shlex.split(cli_cmd))
+        except OSError:
+            cli_cmd = "/usr/local/bin/redis-cli -p {} SHUTDOWN SAVE".format(args.redis_port)
+            subprocess.call(shlex.split(cli_cmd))
+        local_redis.terminate()
+
+    if args.redis_only:
+        logger.warning("Building Redis DB only - no data will be written...")
+        sys.exit(0)
+
+    # Split visibilities into fixed-size chunks
+    freq_chunksize, obj_size = get_freq_chunk(vis.shape, args.obj_size)
+    max_dumps = args.max_dumps if args.max_dumps > 0 else vis.shape[0]
+    # Each vis chunk has a corresponding flags and weights chunk
+    obj_count_per_dump = 3 * vis.shape[1] // freq_chunksize
+    # There is also a timestamps chunk and a weights_channel chunk
+    obj_count = max_dumps * obj_count_per_dump + 2
+    vis_chunk_shape = list(vis.shape)
+    vis_chunk_shape[0] = 1
+    vis_chunk_shape[1] = freq_chunksize
+    vis_chunk_shape = tuple(vis_chunk_shape)
+    ts.add("obj_basename", args.basename)
+    ts.add("obj_vis_chunk_shape", vis_chunk_shape)
+    ts.add("obj_vis_chunk_size", obj_size)
+    ts.add("obj_count", obj_count)
+
+    use_rados = args.ceph_pool is not None
```

Comment:
Minor quibble - if the user specifies both ceph.pool and s3-url, they will only get ceph...

Reply:
Yeah... The problem with putting both into telstate (which makes sense), is which one do you pick when you open telstate since it's all supposed to be "self-contained"... I could go for rados first and then fall back to s3, but is that always a good idea?

Reply:
(I want to avoid having to specify the explicit endpoint of the data in the DataSource, as we discussed).

Reply:
Or we could add a param / query part to the URL that picks one explicitly.

Reply:
I think a default fallback, with the ability to be explicit via the URL when needed should cover all cases.

Reply:
True. That's mostly on the reader side though. On the writer side it would be good to store all possible ways to access the data. But what about writing the data itself? If both s3 and rados point to the same underlying store only the latter is probably needed.